spec/namespaces.md

```diff
@@ -95,6 +107,35 @@ namespace N1.N2
 ```
 the two namespace declarations above contribute to the same declaration space, in this case declaring two classes with the fully qualified names `N1.N2.A` and `N1.N2.B`. Because the two declarations contribute to the same declaration space, it would have been an error if each contained a declaration of a member with the same name.
 
+A *single_line_namespace_declaration* permits a namespace declaration to be written without the `{ ... }` block.  For example:
+
+```csharp
+namespace Name;
+
+extern alias A;
+using B;
+
+class C
+{
+}
+```
+is semantically equivalent to
+```csharp
+namespace Name
+{
+    extern alias A;
+    using B;
+
+    class C
+    {
+    }
+}
+```
+
+Specifically, a *single_line_namespace_declaration* is treated the same as a *namespace_declaration* at the same location in the *compilation_unit* with the same *qualified_identifier*.  The *extern_alias_directive*s, *using_directive*s and *type_declaration*s of that *single_line_namespace_declaration* act as if they were declared in the same order inside the *namespace_body* of that *namespace_declaration*.
+
+A source file cannot contain both a *single_line_namespace_declaration* and a *namespace_declaration*.  A source file cannot contain multiple *single_line_namespace_declaration*s.
+
```

Comment:
note: the grammar enforces this, so this sentence is technically redundant.  But i thought it was still helpful to call out explicitly.  Not sure the spec's perspective on that.